File: python/shout_server.py
import socket

our_socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
PORT = 12345
our_socket.bind(('', PORT))
print('Listening on port', PORT)
while True:
    data, add = our_socket.recvfrom(1024)
    shout = data.decode().upper()
    our_socket.sendto(shout.encode(), add)

File: ocaml/shout_server.ml
open Unix

let socket_server () =
  let sock = socket PF_INET SOCK_DGRAM 0 in
  let server_address = ADDR_INET (inet_addr_any, 12345) in 
  bind sock server_address;
  let buffer = Bytes.create 1024 in 
  while true do
    let n, sender_address = recvfrom sock buffer 0 1024 [] in 
    let received_message = Bytes.sub_string buffer 0 n in
    let reply = String.uppercase_ascii received_message in
    let _ = sendto sock (Bytes.of_string reply) 0 (String.length reply) [] sender_address in
    ()
  done

let _ = socket_server ()
